python/metatensor-operations/tests/unique_metadata.py

```diff
@@ -167,37 +167,31 @@ def test_unique_metadata_block_errors(real_tensor):
     with pytest.raises(TypeError, match=message):
         metatensor.unique_metadata(real_tensor.block(0), "samples", ["structure"])
 
-    message = (
-        "`axis` argument must be a str, either `'samples'` or `'properties'`,"
-        " not <class 'float'>"
-    )
+    message = "`axis` argument must be a str, either `'samples'` or `'properties'`"
     with pytest.raises(TypeError, match=message):
         metatensor.unique_metadata_block(
             real_tensor.block(0),
             axis=3.14,
             names=["structure"],
         )
 
-    message = "`names` argument must be a list of str, not <class 'float'>"
```

Comment:
same comment as other PR, it would be nice to keep the extra context in the error, and disable the check in torch script mode